### ISSUE TITLE
Make SSKR generation flow less confusing

### ISSUE THREAD

**Issue by clouedoc** [aido/app-seed-tool#30]
**Describe the bug**

When opening the app, I'm presented with two choices:

- Check BIP39 recovery phrase
- Check SSKR recovery phrase

My intent is to generate SSKR shards to backup my secret key.

I needed to read the README.md to understand that after entering my BIP39 key, I will be able to generate SSKR shards.

**Expected behavior**

I would like to see a menu item called "Generate SSKR recovery shards".
I would like to not have to type my BIP39 seed (after all, it's supposed to already be inside the device)

**Comment by aido** [comment]
Hi again @clouedoc 

> I would like to not have to type my BIP39 seed (after all, it's supposed to already be inside the device)

Unfortunately this is how the app has to work.
Even though a seed is stored on a device, due to the security of the Ledger devices apps have no way of accessing the seed directly. This is actually a good thing.

To work around this security feature the app has to make a user prove that they know the seed. It uses a little trick to make sure that the user supplied seed can derive the same master key as the onboarded seed. If the derived master keys match then the user supplied seed is correct. Ledger's own "Recovery Check" app uses the exact same technique.
This is also why the app uses the SSKR standard and cannot use  SLIP-39 for share generation. SSKR can do a BIP39 <-> SSKR roundtrip whereas SLIP-39 cannot ... very important

The flow diagram here attempts to explain that a user must first supply the correct BIP39 phrase before they will be offered the option of generating the SSKR shares:

https://github.com/aido/app-seed-tool?tab=readme-ov-file#ledger-nano-application-menus

**Comment by aido** [comment]
Also, when app-seed-tool is released for installation via Ledger Live there will be a blog post written explaining the origins and purpose of the app.

This Ledger blog post has been written but not published yet and it will contain some details on how to use the app too.

**Comment by clouedoc** [comment]
I see, I understand better the technological constraints behind this decision.

From what's left to salvage in terms of ergonomic, I believe that it would be better to have a menu named "generate SSKR shares" that asks for the seed, rather than having a menu named "Check BIP39 seed".

I was confused before reading the diagram, I thought the SSKR Shares generation feature was just missing because I messed up something when building. It took me a minute before thinking about reading the README.md again.

(btw, it was a nice idea to include the diagram, because I'm not sure I would have taken the time to type my BIP39 seed)

**Comment by aido** [comment]
I may add a menu item as you suggest that links back to the same flow as the "Check BIP39 Phase" menu. That may make things less confusing. I am limited to how many menus I can add to Ledger S devices due to the capacity of that device.

**Comment by kjansen387** [comment]
> From what's left to salvage in terms of ergonomic, I believe that it would be better to have a menu named "generate SSKR shares" that asks for the seed, rather than having a menu named "Check BIP39 seed".

I agree, also wasn't clear for me. Better to name it 'generate SSKR shares', as this is the goal of the user, then one sentence explanation that the user has to enter the seed first.

Some other suggestions:
- Let the user check and correct the words. Especially of the entered SSKR shares. It's pretty painful on a nano x (stax/flex are still not supported officially).
- When pressing the 2 buttons on the message 'SSKR verified successfully' message (or something along those lines) the seed tool exits. I should have cycled the menus, but pretty painful as well, as that requires entering the shares again. Better to have that action not exit the app.

**Comment by aido** [comment]
Hi @kjansen387 

> * Let the user check and correct the words. Especially of the entered SSKR shares. It's pretty painful on a nano x (stax/flex are still not supported officially).

While I consider your other suggestions for now I can inform you that v1.8.1 has recently been made available on Ledger Live with support for Stax and Flex. Usability is very much improved on these devices.

**Comment by kjansen387** [comment]
Ah great! Got a Flex specifically to make SSKR less painful, only to find out that it wasn't officially supported. That was on the 18th of August, 9 days ago. Got a bit frustrated, as https://support.ledger.com/article/Seed-tool only lists the Stax as not supported - it doesn't mention the Flex. 

Saw that it could be installed using a more complicated procedure (sideloading). Complained to support, asked them to make it available in Ledger Live. Glad to find out that they did :)

Also, the buttons on the X are not robust enough to do a lot of SSKR on them. 

Just tested it on the Flex, pretty smooth! Enhancement could be to be able to check / correct the words, I think now it's only possible to go back one word. But didn't test with a deliberate mistake. 

Nice work, thanks for creating the app!

**Comment by aido** [comment]
Hi @kjansen387,

> I think now it's only possible to go back one word.

On Flex and Stax I think it might be possible to go back more than just one word. For BIP39 phrases you can go back to start and for SSKR shares you can go back to start of whatever share you are currently inputting.  That's how I think I coded it but its been a while since I tested that particular functionality.

> Also, the buttons on the X are not robust enough to do a lot of SSKR on them.

I totally agree.
But I am glad you are seeing what a big difference the Flex makes when inputting so many words. So much easier.
